tests/sentry/api/endpoints/test_organization_metrics_metadata.py

```diff
@@ -317,19 +317,24 @@ def test_get_no_pre_post_context(self):
         "sentry.sentry_metrics.querying.metadata.metrics_code_locations.CodeLocationsFetcher.MAXIMUM_KEYS",
         50,
     )
-    def test_get_locations_with_too_many_combinations(self):
+    @patch(
+        "sentry.sentry_metrics.querying.metadata.metrics_code_locations.CodeLocationsFetcher._in_batches"
+    )
+    def test_get_locations_with_too_many_combinations(self, _in_batches):
         project = self.create_project(name="project_1")
         mri = "d:custom/sentry.process_profile.track_outcome@second"
 
-        self.get_error_response(
+        self.get_success_response(
             self.organization.slug,
             metric=[mri],
             project=[project.id],
             statsPeriod="90d",
-            status_code=400,
             codeLocations="true",
         )
 
+        args, *_ = _in_batches.call_args
+        assert len(list(args[0])) == 46
```

Comment:
Pretty ugly looking but what we do is to make sure that a function is called with the right generator. Since it's a generator we have to resolve it into a data structure to compute its size.